crates/bevy_reflect/src/array.rs

```diff
@@ -303,8 +300,18 @@ impl Typed for DynamicArray {
 
 /// An iterator over an [`Array`].
 pub struct ArrayIter<'a> {
-    pub(crate) array: &'a dyn Array,
-    pub(crate) index: usize,
+    array: &'a dyn Array,
+    index: usize,
+}
+
+impl<'a> ArrayIter<'a> {
+    /// Creates a new [`ArrayIter`].
+    ///
+    /// It is an [`Iterator`] for [`Array`] whose elements start from `index`.
+    #[inline]
+    pub fn new(array: &'a dyn Array, index: usize) -> ArrayIter {
```

Comment:
As all the actual uses of this Method always provide 0 as the start index, why not hardcode it to that?

Reply:
This would also make it more aligned with other iter types like `FieldIter`

Reply:
I have hardcoded it to 0